src/compute/src/server.rs

```diff
@@ -179,8 +172,38 @@ struct Worker<'w, A: Allocate> {
 }
 
 impl<'w, A: Allocate> Worker<'w, A> {
-    /// Draws from `dataflow_command_receiver` until shutdown.
-    fn run(&mut self) {
+    /// Waits for client connections and runs them to completion.
+    pub fn run(&mut self) {
+        let mut shutdown = false;
+        while !shutdown {
+            match self.client_rx.recv() {
+                Ok((rx, tx)) => self.run_client(rx, tx),
+                Err(_) => shutdown = true,
+            }
+        }
```

Comment:
Nit? loop+break

Reply:
We have elsewhere code that looks like
```rust
        let mut shutdown = false;
        while !shutdown {
            // ...
            // Handle any received commands.
            let mut empty = false;
            while !empty {
                match command_rx.try_recv() {
                    Ok(cmd) => commands.push(cmd),
                    Err(TryRecvError::Empty) => empty = true,
                    Err(TryRecvError::Disconnected) => {
                        empty = true;
                        shutdown = true;
                    }
                }
            }
```
and its the breaks and continues etc to unlabeled `while` loops. In this case, I think the `while !shutdown` condition is great because it explains on what basis we stop looping (pretty clear with so little code, but in larger settings much better than having to scan for the break statements).

Reply:
Works for me! I agree it's more readable when it's not just a tiny loop :)